app/controllers/checkout_controller.rb

```diff
@@ -62,6 +68,22 @@ def update
 
   private
 
+  # Check if items present in cart have been removed from the OC
+  # before checkout has been completed
+  def recheck_stock_availability
```

Comment:
Agree with @enricostano . This belongs to a model validation that adds ` t('checkout.items_removed_from_oc')` in its model errors.